src/prefect/tasks/database/sqlite.py

```diff
@@ -81,7 +95,12 @@ def run(self, db: str = None, script: str = None):
             - script (str, optional): the optional script to execute at runtime;
                 if not provided, `self.script` will be used instead.
         """
+        if db is None:
+            raise ValueError("The db must be specified")
+
+        if script is None:
+            raise ValueError("The query script string must be specified")
+
         with closing(sql.connect(db)) as conn:
             with closing(conn.cursor()) as cursor:
                 cursor.executescript(script)
-                conn.commit()
```

Comment:
I think we need to keep this call. Based on the [`sqlite3` docs](https://docs.python.org/3/library/sqlite3.html#sqlite3.Connection.commit), changes aren't saved to the database unless `commit` is called.

Reply:
I removed the `os.remove` from https://github.com/ahuang11/prefect-task-verification/blob/main/database-tasks/sqlite.py and ran the script.

The changes seem committed to me. I think it's because it's wrapped in the `with` statement?
https://stackoverflow.com/a/19522634

```
(prefect-v1) [Thu 31 10:23] database-tasks: python sqlite.py
[2022-03-31 10:23:35-0600] INFO - prefect.FlowRunner | Beginning Flow run for 'sqlite'
[2022-03-31 10:23:35-0600] INFO - prefect.TaskRunner | Task 'SQLiteScript': Starting task run...
[2022-03-31 10:23:35-0600] INFO - prefect.TaskRunner | Task 'SQLiteScript': Finished task run for task with final state: 'Success'
[2022-03-31 10:23:35-0600] INFO - prefect.TaskRunner | Task 'SQLiteQuery': Starting task run...
[2022-03-31 10:23:35-0600] INFO - prefect.TaskRunner | Task 'SQLiteQuery': Finished task run for task with final state: 'Success'
[2022-03-31 10:23:35-0600] INFO - prefect.TaskRunner | Task '_show_output': Starting task run...
None
[2022-03-31 10:23:35-0600] INFO - prefect.TaskRunner | Task '_show_output': Finished task run for task with final state: 'Success'
[2022-03-31 10:23:35-0600] INFO - prefect.TaskRunner | Task '_show_output': Starting task run...
[('Prefection', 'Prefectionist', 2022)]
[2022-03-31 10:23:35-0600] INFO - prefect.TaskRunner | Task '_show_output': Finished task run for task with final state: 'Success'
[2022-03-31 10:23:35-0600] INFO - prefect.FlowRunner | Flow run SUCCESS: all reference tasks succeeded
(prefect-v1) [Thu 31 10:23] database-tasks: ls -ltrh
total 32
-rw-r--r--  1 andrew  staff   1.2K Mar 31 10:22 sqlite.py
-rw-r--r--  1 andrew  staff    12K Mar 31 10:23 prefect.db
(prefect-v1) [Thu 31 10:23] database-tasks: sqlite3 prefect.db
SQLite version 3.37.1 2021-12-30 15:30:28
Enter ".help" for usage hints.
sqlite> SELECT * FROM book;
Dirk Gently's Holistic Detective Agency|Douglas Adams|1987
Prefection|Prefectionist|2022
```

Reply:
Yes, you're right. The important bit is tucked way down at the bottom: https://docs.python.org/3/library/sqlite3.html#using-the-connection-as-a-context-manager. So the combination of `closing` and using the connection as a context manager makes sure that the operations are committed and the connection is closed. TIL.